github-actions/lock-closed/src/main.ts

```diff
@@ -0,0 +1,78 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+
+async function lockIssue(client: github.GitHub, issue: number, message?: string): Promise<void> {
+  // Add a comment before locking
+  if (message) {
+    await client.issues.createComment({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+      issue_number: issue,
+      body: message,
+    });
+  }
+
+  // Actually lock the issue
+  await client.issues.lock({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: issue,
+  });
+}
+
+async function run(): Promise<void> {
+  try {
+    // NOTE: `days` and `message` must not be changed without dev-rel and dev-infra concurrence
+
+    // Fixed amount of days a closed issue must be inactive before being locked
+    const days = 30;
+    // Standardized Angular Team message for locking issues
+    const message =
+      'This issue has been automatically locked due to inactivity.\n' +
+      'Please file a new issue if you are encountering a similar or related problem.\n' +
+      '\n' +
+      'Automatically locking inactive issues helps keep GitHub notifications relevant for project maintainers.\n' +
```

Comment:
Rather than this line, I think we should link to the policy doc being created in https://github.com/angular/angular/pull/32405

```
Read more about our [automatic conversation locking policy](https://github.com/angular/angular/{sha}/docs/GITHUB_PROCESS.md#automatic-conversation-locking)
```

Reply:
Done.